agent-bootstrap/src/main/java/io/opentelemetry/auto/bootstrap/instrumentation/decorator/ClientDecorator.java

```diff
@@ -15,10 +15,51 @@
  */
 package io.opentelemetry.auto.bootstrap.instrumentation.decorator;
 
+import io.grpc.Context;
+import io.opentelemetry.context.ContextUtils;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.trace.DefaultSpan;
 import io.opentelemetry.trace.Span;
+import io.opentelemetry.trace.Span.Kind;
+import io.opentelemetry.trace.Tracer;
+import io.opentelemetry.trace.TracingContextUtils;
 
 public abstract class ClientDecorator extends BaseDecorator {
 
+  // Keeps track of the client span in a subtree corresponding to a client request.
+  private static final Context.Key<Span> CONTEXT_CLIENT_SPAN_KEY =
+      Context.key("opentelemetry-trace-auto-client-span-key");
+
+  public static Scope currentContextWith(final Span clientSpan) {
+    if (!clientSpan.getContext().isValid()) {
+      return TracingContextUtils.currentContextWith(clientSpan);
+    }
+    return ContextUtils.withScopedContext(
+        TracingContextUtils.withSpan(
+            clientSpan, Context.current().withValue(CONTEXT_CLIENT_SPAN_KEY, clientSpan)));
+  }
+
+  public static Context withSpan(final Span clientSpan, final Context context) {
```

Comment:
It looks like this is only called with `Context.current()`, so maybe this method isn't needed?

Reply:
I noticed the same lack of symmetry in `TracingContextUtils` and followed the pattern. Let me know if it's worth diverging.

Reply:
`io.opentelemetry.trace.TracingContextUtils#withSpan` can potentially be called with contexts other than current during context extraction via `io.opentelemetry.trace.propagation.HttpTraceContext#extract` method. This client decorator is not general purpose util, but more specific one. Thus I think we indeed can merge these two methods together.